point/gemini_commands.py

```diff
@@ -104,26 +106,45 @@ def parse_revisions(string):
     return vals
 
 
+def parse_ip4vaddr(string):
+    match = _re_ipv4addr.fullmatch(string)
+    if match is None:                                        raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(1)) < 0 or int(match.group(1)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(2)) < 0 or int(match.group(2)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(3)) < 0 or int(match.group(3)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(4)) < 0 or int(match.group(4)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    return ipaddress.IPv4Address(string)
+
 ####################################################################################################
 
 
-# returns tuple: (int:sign[-1|0|+1], int:deg, int:min, float:sec)
+# returns tuple: (int:sign[-1|0|+1], int:hour, int:min, int:sec)
+def ang_to_hourminsec(ang):
+    return ang_to_degminsec(ang * 24.0 / 360.0)
+
+# returns tuple: (int:sign[-1|0|+1], int:deg, int:min, int:sec)
 def ang_to_degminsec(ang):
     if   ang > 0.0: sign = +1.0
     elif ang < 0.0: sign = -1.0
     else:           sign =  0.0
     ang = abs(ang) * 3600.0
-    f_sec = ang % 60.0
+    i_sec = int(ang % 60.0) # TODO: change this to round(), if we can fix the round-up-to-60 issues
     ang /= 60.0
     i_min = int(ang % 60.0)
     ang /= 60.0
     i_deg = int(ang)
-    return (sign, i_deg, i_min, f_sec)
-# TODO: test this, in case I am dumb
+    return (sign, i_deg, i_min, i_sec)
 
-# returns tuple: (int:sign[-1|0|+1], int:hour, int:min, float:sec)
-def ang_to_hourminsec(ang):
-    return ang_to_degminsec(ang * 24.0 / 360.0)
+# returns tuple: (int:sign[-1|0|+1], int:deg, int:min)
+def ang_to_degmin(ang):
+    if   ang > 0.0: sign = +1.0
+    elif ang < 0.0: sign = -1.0
+    else:           sign =  0.0
```

Comment:
Do we actually need a special case for `ang == 0`? If not, could compact this to `sign = +1 if ang >= 0 else -1`

Reply:
IIRC LX200 command documentation freaked me out slightly about sign characters way back when first implementing these conversion functions, so I just elected to be as safe and deliberate as possible here. I guess in all current cases, I do ultimately end up doing e.g. `self._signchar = '+' if sign >= 0.0 else '-'` in the downstream code. I dunno. :man_shrugging: